volttron/platform/dbutils/postgresqlfuncts.py

```diff
@@ -350,7 +350,7 @@ def get_topic_meta_map(self):
             'SELECT topic_id, metadata '
             'FROM {}').format(Identifier(self.meta_table))
         rows = self.select(query)
-        meta_map = {tid: jsonapi.loads(meta) if meta else None for tid, meta in rows}
+        meta_map = {tid: meta if meta and isinstance(meta, dict) else jsonapi.loads(meta) if meta else None for tid, meta in rows}
```

Comment:
Just for my understanding, is this nested ternary expression one-liner equivalent to the following for-loop?

```python
meta_map = {}
for tid, meta in rows:
   if meta:
       if meta and isinstance(meta, dict):
           meta_map[tid] = meta
       else:
           meta_map[tid] = jsonapi.loads(meta) 
   else:
       meta_map[tid] = None
```

Reply:
Yes

Reply:
Just curious. Is obfuscation best practice for this project?

Reply:
@awoz No it's not

I think nested if looping this way is harder to read as well.  The original meta_map was the right level of complexity, but I did have a hard time following the new code.

@ntenney can you replace the code with something more user friendly.

Reply:
> @awoz No it's not
> 
> I think nested if looping this way is harder to read as well. The original meta_map was the right level of complexity, but I did have a hard time following the new code.
> 
> @ntenney can you replace the code with something more user friendly.

@ntenney If it helps, feel free to copy and/or modify the `for loop` in my initial comment above ^^

Reply:
I'm not a huge fan of splitting something that takes 1 line into 9. Would this be more palatable?

`meta_map = {tid: meta if meta and type(meta) == dict else (json.loads(meta) if meta else None) for tid, meta in rows}`

Reply:
@ntenney splitting the long line or adding the parenthesis doesn't make it any clearer for me. I do get what the code is doing but I agree that the broken down for loop was much easier to read.  I might have voted for the compact format if there was just a single if inside but the second if inside for loop inside the {} really pushes it over the edge for me

Reply:
It is my general sense that all source code should be written "for other" developers who may not be familiar with your code. For them to be successful in their roles, source code hints in the form of comments or clean format are essential. This principle allows for an efficient coding, debug and review cycle, with a timely delivery of solutions. I'm pretty sure that understanding and debugging / single stepping a one-liner is not very palatable.